src/SFML/Graphics/ImageLoader.cpp

```diff
@@ -238,40 +239,39 @@ bool ImageLoader::loadImageFromStream(InputStream& stream, std::vector<Uint8>& p
 
 
 ////////////////////////////////////////////////////////////
-bool ImageLoader::saveImageToFile(const std::string& filename, const std::vector<Uint8>& pixels, const Vector2u& size)
+bool ImageLoader::saveImageToFile(const std::filesystem::path& filename, const std::vector<Uint8>& pixels, const Vector2u& size)
 {
     // Make sure the image is not empty
     if (!pixels.empty() && (size.x > 0) && (size.y > 0))
     {
         // Deduce the image type from its extension
 
         // Extract the extension
-        const std::size_t dot = filename.find_last_of('.');
-        const std::string extension = dot != std::string::npos ? toLower(filename.substr(dot + 1)) : "";
+        const std::string extension = filename.extension().string();
```

Comment:
Does it need to be a  `std::string`?

Reply:
A quick local test shows that `extension` could be a `std::filesystem::path` and the comparisons to character arrays on later lines would still compile and presumably behave the same.